src/Data/Pool/Internal.hs

```diff
@@ -116,18 +137,49 @@ newPool pc = do
               , reaperRef  = ref
               }
   where
-    quotCeil :: Int -> Int -> Int
-    quotCeil x y =
-      -- Basically ceiling (x / y) without going through Double.
-      let (z, r) = x `quotRem` y in if r == 0 then z else z + 1
-
     -- Collect stale resources from the pool once per second.
     collector pools = forever $ do
       threadDelay 1000000
       now <- getMonotonicTime
       let isStale e = now - lastUsed e > poolCacheTTL pc
       mapM_ (cleanStripe isStale (freeResource pc) . stripeVar) pools
 
+-- | A datatype representing the requested maximum resources and count of
+-- stripes. We don't use these figurs directly, but instead calculate  a
+data Input = Input { inputMaxResources :: !Int, inputStripes :: !Int }
+  deriving Show
+
+data StripeResourceAllocation = StripeResourceAllocation { poolInput :: !Input, allowedStripes :: !Int }
+  deriving Show
+
+-- | Determine how many resources should be allocated to each stripe.
+--
+-- The output list contains a single `Int` per stripe, with the 'Int'
+-- representing the amount of resources available to that stripe.
+robin :: StripeResourceAllocation -> [Int]
+robin stripeResourceAllocation = do
+  let counts = replicate (allowedStripes stripeResourceAllocation) 0
+  go (inputMaxResources (poolInput stripeResourceAllocation)) [] counts
+  where
+    go remainingResources done todo = do
+      if remainingResources > 0
+      then
+        case todo of
+          [] ->
+            go remainingResources [] done
+          (r : rs) -> do
+            go (remainingResources - 1) (r + 1 : done) rs
+      else done ++ todo
```

Comment:
i'm like 100% confident there's an easy way to do this with math but i'm a programmer not a mathematician

Reply:
How abount something like this?
```haskell
robin :: StripeResourceAllocation -> [Int]
robin stripeResourceAllocation =
  let numStripes = allowedStripes stripeResourceAllocation
      (baseCount, remainder) = inputMaxResources (poolInput stripeResourceAllocation) `divMod` numStripes
  in replicate remainder (baseCount + 1) ++ replicate (numStripes - remainder) baseCount
```

Reply:
Thanks! That's a good approach 😄